source/image-handler/image-request.js

```diff
@@ -136,9 +136,7 @@ class ImageRequest {
             const decoded = this.decodeRequest(event);
             return decoded.key;
         } else if (requestType === "Thumbor" || requestType === "Custom") {
-            // Parse the key from the end of the path
-            const key = (event["path"]).split("/");
-            return key[key.length - 1];
+            return decodeURIComponent(event["path"].replace(/\d+x\d+\/|filters[:-][^/;]+|\/fit-in\/+|^\/+/g, '').replace(/^\/+/, ''));
```

Comment:
Changed

Reply:
That's different from my fix in aws-manage, I'm wondering if it's more or less robust 🤔 

Mine is very tied to our usage, I take the path coming after the last `)`. Does yours cover all the transformations?